test/dlc_tests/sanity/test_pre_release.py

```diff
@@ -197,6 +208,71 @@ def test_emacs(image):
     _run_cmd_on_container(container_name, ctx, "emacs -version")
 
 
+@pytest.mark.model("N/A")
+def test_cuda_paths(gpu):
+    """
+    Test to ensure directory structure for GPU files has cuda version in it
+
+    :param gpu: gpu image uris
+    """
+    image = gpu
+    if "example" in image:
+        pytest.skip("Skipping example containers which are not explicitly tied to a cuda version")
+
+    dlc_path = os.getcwd().split('/test/')[0]
+    job_type = "training" if "training" in image else "inference"
+
+    # Ensure that image has a supported framework
+    frameworks = ("tensorflow", "pytorch", "mxnet")
+    framework = ''
+    for fw in frameworks:
+        if fw in image:
+            framework = fw
+            break
+    assert framework, f"Cannot find any frameworks {frameworks} in image uri {image}"
+
+    # Get cuda and framework version through regex
+    cuda_version_match = re.search(r'-cu\d+-', image)
+    cuda_version = cuda_version_match.group().strip('-')
```

Comment:
Generally, you don't have to write two lines of code to do a regex query on a string, especially if all you want is the value you are looking for (as opposed to all the other information on the `re.Match` object).

You can do:
```
cuda_version = re.search(...).group().strip("-")
```

Only downside to writing this in a single line is when your search/match fails, and you get a `NoneType` output instead of `re.Match` from the `re.search` function call. In that case, you would want to throw an Exception anyway.

Also, you can use groups, so that you wouldn't need to use the `.strip("-")` here.

Reply:
I want to avoid doing too much string manipulation in one line for readability.

Reply:
Sure. Makes sense to do that for readability.

Please do consider using regex match-groups, though.

Reply:
updated